packages/teams-js/src/private/associatedApps.ts

```diff
@@ -0,0 +1,237 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ErrorCode, SdkError, TabInstance } from '../public';
+import { runtime } from '../public/runtime';
+
+// Open questions
+// 1. I've re-used `TabInstance` from the public API, does that contain all of the information you and app developers might need?
```

Comment:
That's great! We should double check the type since it's been a long time since this was created.

Reply:
Yup, please take a look! It looks like it could work, and since it's whatis returned by `pages.tabs.getTabInstances` there's some nice synergy.

Reply:
Looked at the object thats returne" [](https://github.com/OfficeDev/microsoft-teams-library-js/blob/931a483ea20f116f6d6f7ca407cea3c03192e1fb/packages/teams-js/src/public/interfaces.ts#L17). It does contain what we need

Reply:
The object that is returned has a lot more fields compared to this. This is missing order, removeUrl, appId. I think we would need to create a new object.